node/cn/backend.go

```diff
@@ -272,6 +272,10 @@ func New(ctx *node.ServiceContext, config *Config) (*CN, error) {
 		return nil, err
 	}
 	bc.SetCanonicalBlock(config.StartBlockNumber)
+	if chainConfig.IsKoreForkEnabled(bc.CurrentBlock().Number()) {
```

Comment:
@hyunsooda We cannot modify the genesis parameters in the production environment. (see https://github.com/klaytn/klaytn/blob/81db7606f5252d8fa9d9cc0ad4769a0ee182220b/blockchain/genesis.go#L270).

I agree removing duplicate deriveShaImpl initialization. `cn.New` is a great location for chain config modification:
https://github.com/klaytn/klaytn/blob/81db7606f5252d8fa9d9cc0ad4769a0ee182220b/node/cn/backend.go#L222

Reply:
Test passed when I moved `IsKoreForkEnabled` function as you mentioned.
Let's have a future conversation where to put this logic. +`homi` issue

Reply:
Hi @ian0371,

1. I meant that we may make a change so that the `genesis.json` output of homi will write zero value to `deriveShaImpl` field by default. 
2. `if chainConfig.IsKoreForkEnabled(bc.CurrentBlock().Number()) {` This dependency requires setup blockchain instance before chain configuration init. Its instantiation is done at `NewBlockChain()` and I suggested inserting inside of it (near `bc.loadLaststate()` call). 

Let me know if I’m missing something.

Reply:
@hyunsooda 
1. `homi setup --cn-num 1` will already generate `genesis.json` with `deriveShaImpl=0`. If you meant `homi setup --cypress` should also generate `genesis.json` with `deriveShaImpl=0`, I'm not sure if we should do that because if it's used on the production environment, `kcn init` will fail due to the conflict with the existing `deriveShaImpl=2` from database.
2. You're right, we need an initialized blockchain instance. Your suggested location seems fine.

Reply:
@ian0371,

1. I found another reason that we should not modify the target of cypress `genesis.json` because the newbie is not able to sync due to the failure of verification by unmatched tx root. Because the very initial blocks include tx roots calculated with `concat` method.

This is just to clarify your explanation, Is there a condition that asserts `deriveShaImpl` method must not be modified when the `genesis.json` is initialized again? I couldn't find where it is and checked out that reinitialization was successfully done and ran.

Thanks.

Reply:
@hyunsooda 
I tested re-initialization with different deriveShaImpl and failed with the following error.
```
$ homi setup local --cypress --cn-num 1
$ bin/kcn --datadir ./data init ./data/genesis.json
... success
$ # modify genesis.json: deriveShaImpl = 2 -> 0
$ bin/kcn --datadir ./data init ./data/genesis.json
...
CRIT[11/02,13:32:12 +09] [19] Failed to write genesis block             err="database already contains an incompatible genesis block (have 9c8c16594f977841, new 1a67dab39e7e7afd)"
```

The checking logic is here: 
https://github.com/klaytn/klaytn/blob/507698b99c32551721281f85e5be1a629610f8f3/blockchain/genesis.go#L227-L230

How did you test to make it not fail?

Reply:
I've used a custom script that removes data dir and runs `init` command for easier testing and used it when I tested. Sorry for the noise and thanks for the ref.

Reply:
@ian0371 maybe you didn't removed chaindata after you made a genesis block.

Reply:
@hqjang-pepper Yep, the failure is intended. It's to show that we need to be careful when changing the default value of `genesis.json` generated from homi.